diff --git a/lua/doom/extras/keybindings/init.lua b/lua/doom/extras/keybindings/init.lua
index 5f97ad1..eedecca 100644
--- a/lua/doom/extras/keybindings/init.lua
+++ b/lua/doom/extras/keybindings/init.lua
@@ -252,7 +252,7 @@ else
 end
 
 -- TAB to cycle buffers too, why not?
-utils.map("n", "<Tab>", ":bnext<CR>", opts, "Movement", "cycle_next_buffer", "Goto next buffer")
+--utils.map("n", "<Tab>", ":bnext<CR>", opts, "Movement", "cycle_next_buffer", "Goto next buffer")
 utils.map(
   "n",
   "<S-Tab>",
@@ -574,6 +574,15 @@ utils.map(
   "prev_buffer_alt",
   "Goto previous buffer"
 )
+utils.map(
+  "n",
+  "<leader><tab>",
+  '<cmd>lua require("bufferline").cycle(-1)<CR>',
+  opts,
+  "Buffer",
+  "prev_buffer_alt2",
+  "Goto previous buffer"
+)
 utils.map(
   "n",
   "<leader>bp",
@@ -1156,3 +1165,127 @@ utils.map("n", "<leader>m", "<cmd>w<cr>", opts, "Save", "save_right", "Save m")
 
 -- man pages
 utils.map("n", "<leader>h", ":Man ", { silent = false }, "Man page", "man_page", "Man page")
+
+-- MINE
+utils.map(
+  "n",
+  "<leader>op",
+  "<cmd>lua require'telescope'.extensions.project.project{}<CR>",
+  opts,
+  "Editor",
+  "open_project",
+  "Open project"
+)
+utils.map(
+  "n",
+  "<leader>*",
+  "<cmd>lua my_open_tele()<cr>",
+  opts,
+  "Editor",
+  "search_word",
+  "Search word under cursor, raw"
+)
+utils.map(
+  "n",
+  "<leader>sr",
+  "<cmd>lua require('telescope').extensions.live_grep_raw.live_grep_raw()<cr>",
+  opts,
+  "Editor",
+  "search_text_raw",
+  "Search text raw"
+)
+
+utils.map(
+  "n",
+  "<leader>wm",
+  "<cmd>lua maximize()<cr>",
+  opts,
+  "Editor",
+  "maximize",
+  "Maximize window"
+)
+utils.map(
+  "n",
+  "<leader>gt",
+  "<cmd>lua require'agitator'.git_time_machine()<cr>",
+  opts,
+  "Editor",
+  "git_time_machine",
+  "Open project"
+)
+utils.map(
+  "n",
+  "<leader>gB",
+  "<cmd>lua require'agitator'.git_blame_toggle()<cr>",
+  opts,
+  "Editor",
+  "git_blame",
+  "Open project"
+)
+utils.map(
+  "n",
+  "<leader>gf",
+  "<cmd>lua require'agitator'.open_file_git_branch()<cr>",
+  opts,
+  "Editor",
+  "open_file_git_branch",
+  "Open project"
+)
+utils.map(
+  "n",
+  "<leader>gd",
+  "<cmd>lua require'diffview'.open()<cr>",
+  opts,
+  "Editor",
+  "diffview",
+  "Open project"
+)
+utils.map(
+  "n",
+  "<leader>gL",
+  "<cmd>lua require'diffview'.file_history('.')<cr>",
+  opts,
+  "Editor",
+  "project_history",
+  "Open project"
+)
+utils.map(
+  "n",
+  "<leader>gT",
+  "<cmd>lua require'diffview'.file_history()<cr>",
+  opts,
+  "Editor",
+  "file_history",
+  "Open project"
+)
+utils.map(
+  "n",
+  "<leader>gg",
+  "<cmd>Neogit<cr>",
+  opts,
+  "Editor",
+  "neogit",
+  "Open project"
+)
+
+-- https://stackoverflow.com/a/19619038/516188
+-- if that doesn't cut it, consider https://github.com/qpkorr/vim-bufkill
+utils.map(
+  "n",
+  "<leader>bd",
+  ":b#<bar>bd#<CR>",
+  opts,
+  "Editor",
+  "delete_buffer",
+  "Delete buffer"
+)
+
+utils.map(
+  "n",
+  "<leader>td",
+  ":tabc<cr>",
+  opts,
+  "Editor",
+  "delete_tab",
+  "Delete tab"
+)
diff --git a/lua/doom/modules/config/doom-telescope.lua b/lua/doom/modules/config/doom-telescope.lua
index 06e0b1e..4edd4ae 100644
--- a/lua/doom/modules/config/doom-telescope.lua
+++ b/lua/doom/modules/config/doom-telescope.lua
@@ -6,6 +6,9 @@ return function()
 
   telescope.setup({
     defaults = {
+      -- START emmanuel
+      path_display = {'truncate'},
+      -- END emmanuel
       find_command = {
         "rg",
         "--no-heading",
@@ -62,6 +65,8 @@ return function()
           ["<C-q>"] = actions.smart_send_to_qflist + actions.open_qflist,
           ["<Leader>f"] = actions.close, -- works like a toggle, sometimes can be buggy
           ["<CR>"] = actions.select_default + actions.center,
+          ["<C-n>"] = actions.cycle_history_next,
+          ["<C-p>"] = actions.cycle_history_prev,
         },
         n = {
           ["<C-j>"] = actions.move_selection_next,
diff --git a/lua/doom/modules/config/doom-whichkey.lua b/lua/doom/modules/config/doom-whichkey.lua
index ec5bada..091fca7 100644
--- a/lua/doom/modules/config/doom-whichkey.lua
+++ b/lua/doom/modules/config/doom-whichkey.lua
@@ -199,6 +199,7 @@ return function()
       ["m"] = { "Minimap" },
       ["s"] = { "Symbols" },
       ["t"] = { "Terminal" },
+      ["p"] = { "Project" }, -- E
     },
     ["j"] = {
       name = "+jumps",
diff --git a/lua/doom/modules/init.lua b/lua/doom/modules/init.lua
index 04bf3c6..435199e 100644
--- a/lua/doom/modules/init.lua
+++ b/lua/doom/modules/init.lua
@@ -144,7 +144,8 @@ packer.startup(function(use)
   -- can be disabled to use your own statusline
   local disabled_statusline = functions.is_plugin_disabled("statusline")
   use({
-    "glepnir/galaxyline.nvim",
+    "Unoqwy/galaxyline.nvim",
+    branch = 'short-buftypes-inactive',
     config = require("doom.modules.config.doom-eviline"),
     disable = disabled_statusline,
     event = "ColorScheme",
@@ -310,6 +311,7 @@ packer.startup(function(use)
     config = require("doom.modules.config.doom-lspconfig"),
     disable = disabled_lsp,
     event = "ColorScheme",
+    commit = '2e6c94069e92cb4a43416b3ac0267e941b3fd47e'
   })
 
   -- Completion plugin
