diff --git a/lua/doom/modules/config/doom-telescope.lua b/lua/doom/modules/config/doom-telescope.lua
index 06e0b1e..49912e7 100644
--- a/lua/doom/modules/config/doom-telescope.lua
+++ b/lua/doom/modules/config/doom-telescope.lua
@@ -6,6 +6,9 @@ return function()
 
   telescope.setup({
     defaults = {
+      -- START emmanuel
+      path_display = {'truncate'},
+      -- END emmanuel
       find_command = {
         "rg",
         "--no-heading",
